File: src/math/problems/SumOfDigits.java
package src.math.problems;

import src.core.Problem;
import src.core.TestCase;

public class SumOfDigits extends Problem<Integer, Integer> {

    public static void main(String[] args) {
        new SumOfDigits().run();
    }

    @Override
    protected TestCase<Integer, Integer>[] getTestCases() {
        return new TestCase[]{
                new TestCase(123, 6),
                new TestCase<>(89723478, 48)
        };
    }

    @Override
    protected Integer solve(Integer testCaseInput) {
        return reverseDigits(testCaseInput);
    }

    private int reverseDigits(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }
}

File: src/math/problems/SumOfDigits.kt
package src.math.problems

import src.core.Problem
import src.core.TestCase

class SumOfDigits : Problem<Int, Int>() {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) = SumOfDigits().run()
    }

    override fun getTestCases(): Array<TestCase<Int, Int>> = arrayOf(
        TestCase(123, 6),
        TestCase(89723478, 48)
    )

    override fun solve(testCaseInput: Int): Int {
        return reverseDigits(testCaseInput)
    }

    private fun reverseDigits(n: Int): Int {
        var x = n
        var sum = 0
        while (x > 0) {
            sum += x % 10
            x /= 10
        }
        return sum
    }
}
